docker.go

```diff
@@ -718,7 +720,8 @@ func (c *DockerContainer) StartLogProducer(ctx context.Context) error {
 // and sending them to each added LogConsumer
 func (c *DockerContainer) StopLogProducer() error {
 	if c.stopProducer != nil {
-		c.stopProducer <- true
+		// Cancel the producer's context.
+		c.stopProducer()
```

Comment:
I think there is no `stopProducer()` method at the container level

Reply:
I changed the channel to a CancelFunc in the latest version:
```diff
-       stopProducer      chan bool
+       stopProducer      context.CancelFunc
```

Reply:
Oh sorry, I looked at my open IDE instead 🤦